### PR TITLE
Make GetPluginConfig accessible from other packages.

### LINKED ISSUE #4865: Expose GetPluginConfig
This hasn't moved to pkg/plugin/framework per `TODO:` comment
https://github.com/vmware-tanzu/velero/blob/04345af17a13314765adce5df3531c733a175aea/pkg/restore/restic_restore_action.go#L259-L261
function also need to be accessible from other go packages by renaming function to `GetPluginConfig`

### PR DESCRIPTION
**Pull request by kaovilai**
…n/framework/common/plugin_config.go/GetPluginConfig

Fixes #4865 Expose GetPluginConfig

Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #(issue)

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

File: pkg/restore/pod_volume_restore_action.go [hunk 1]
```diff
@@ -260,35 +259,6 @@ func getSecurityContext(log logrus.FieldLogger, config *corev1.ConfigMap) (strin
 		config.Data["secCtx"]
 }
 
-// TODO eventually this can move to pkg/plugin/framework since it'll be used across multiple
-// plugins.
-func getPluginConfig(kind common.PluginKind, name string, client corev1client.ConfigMapInterface) (*corev1.ConfigMap, error) {
-	opts := metav1.ListOptions{
-		// velero.io/plugin-config: true
-		// velero.io/pod-volume-restore: RestoreItemAction
-		LabelSelector: fmt.Sprintf("velero.io/plugin-config,%s=%s", name, kind),
```
Comment (kaovilai) [inline pkg/restore/pod_volume_restore_action.go:269]: Not sure if this was a bug or not, but to get unit test to pass, I had to add `=true` like [here](https://github.com/vmware-tanzu/velero/pull/6151/files#diff-bfb795c09df77918f0d92045cb9792eeae1d73051311ea3b331097398bb86981R30)

Comment (qiuming-best) [reply pkg/restore/pod_volume_restore_action.go:269]: From velero source codes, also I couldn't see any significance of this variable. But plugins also use this label-selector, so it is better to retain this variable.

**Review by qiuming-best — APPROVED** [review]

**Review by blackpiglet — APPROVED** [review]

File: pkg/plugin/framework/common/plugin_config.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	"fmt"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+func PluginConfigLabelSelector(kind PluginKind, name string) string {
+	return fmt.Sprintf("velero.io/plugin-config=true,%s=%s", name, kind)
```
Comment (danfengliu) [inline pkg/plugin/framework/common/plugin_config.go:30]: Regarding feature [Changing PVC selected-node](https://velero.io/docs/v1.11/restore-reference/#changing-pvc-selected-node:~:text=the%20second%20%22dev%22.-,Changing%20PVC%20selected%2Dnode,-Velero%20can%20update), `plugin-config` should not set to true as a const.

Comment (danfengliu) [reply pkg/plugin/framework/common/plugin_config.go:30]: @kaovilai could you please take a look here

File: pkg/restore/pod_volume_restore_action.go (hunk 1 above)
Comment (qiuming-best) [reply pkg/restore/pod_volume_restore_action.go:269]: @kaovilai Sorry for my misunderstanding, you needn't add `=true`, In fact, `velero.io/plugin-config` will pattern any values with the key. No matter whether the value is empty or is true. which means that if you're using `velero.io/plugin-config`, the UT could still be passed.

But now if you change into `velero.io/plugin-config=true`, It only could pattern those resources have `velero.io/plugin-config=true`, which will make the [instruction](https://velero.io/docs/v1.11/restore-reference/#changing-pvc-selected-node:~:text=the%20second%20%22dev%22.-,Changing%20PVC%20selected%2Dnode,-Velero%20can%20update) in the doc will not work

**Pull request merged**